src/main/java/com/whiteclarkegroup/liquibaselinter/config/rules/RuleRunner.java

```diff
@@ -84,45 +91,42 @@ public void checkChangeLog(DatabaseChangeLog databaseChangeLog) throws ChangeLog
         for (ChangeLogRule changeLogRule : changeLogRules) {
             final List<RuleConfig> configs = config.forRule(changeLogRule.getName());
             for (RuleConfig ruleConfig : configs) {
-                changeLogRule.configure(ruleConfig);
-                if (ConditionHelper.evaluateCondition(ruleConfig, databaseChangeLog) && changeLogRule.invalid(databaseChangeLog)) {
-                    handleViolation(changeLogRule.getMessage(databaseChangeLog), changeLogRule.getName(), ruleConfig, databaseChangeLog, null);
+                if (isEnabled(ruleConfig)) {
+                    changeLogRule.configure(ruleConfig);
+                    if (ConditionHelper.evaluateCondition(ruleConfig, databaseChangeLog) && changeLogRule.invalid(databaseChangeLog)) {
+                        handleViolation(changeLogRule.getMessage(databaseChangeLog), changeLogRule.getName(), databaseChangeLog, null);
+                    }
                 }
             }
         }
     }
 
-    private void handleViolation(String errorMessage, String rule, RuleConfig ruleConfig, DatabaseChangeLog databaseChangeLog, ChangeSet changeSet) throws ChangeLogParseException {
-        if (!isEnabledAfter(ruleConfig) || isIgnored(rule, changeSet)) {
+    private void handleViolation(String errorMessage, String rule, DatabaseChangeLog databaseChangeLog, ChangeSet changeSet) throws ChangeLogParseException {
+        if (isIgnored(rule, changeSet)) {
             report.addIgnored(databaseChangeLog, changeSet, rule, errorMessage);
-            return;
-        }
-        if (config.isFailFast()) {
+        } else if (config.isFailFast()) {
             throw ChangeLogParseExceptionHelper.build(databaseChangeLog, changeSet, errorMessage);
         } else {
             report.addError(databaseChangeLog, changeSet, rule, errorMessage);
         }
     }
 
     private boolean isIgnored(String ruleName, ChangeSet changeSet) {
-        if (changeSet == null || changeSet.getComments() == null || !changeSet.getComments().contains(LQL_IGNORE_TOKEN)) {
-            return false;
-        }
-        final String comments = changeSet.getComments();
-        final String toIgnore = comments.substring(comments.indexOf(LQL_IGNORE_TOKEN) + LQL_IGNORE_TOKEN.length());
-        final String[] split = toIgnore.split(",");
-        return Arrays.stream(split).anyMatch(ruleName::equalsIgnoreCase);
-    }
-
-    private boolean isEnabledAfter(RuleConfig ruleConfig) {
-        if (!config.isEnabledAfter() && !ruleConfig.isEnabledAfter()) {
+        final String comments = Optional.ofNullable(changeSet).map(ChangeSet::getComments).orElse("");
+        if (comments.endsWith(LQL_IGNORE_TOKEN)) {
```

Comment:
By moving `lql-ignore` comment suffix checks here, rule violations are now reported as ignored. Before they were never evaluated. The philosophy I'm presenting here is:
* `lqlint.json` can cause rule evaluations to be skipped (never evaluated)
* `changelog.xml` comments can cause rule violations to be ignored, but not skipped.

Reply:
Yeah I do think this makes sense. The config rule has blanket rules based on a point in history or a context representing a bunch of stuff that just shouldn't be checked as it's not applicable. Where an ignore comment is a break-glass thing - we had to break the rule.

@luke-rogers what do you think?

Reply:
Looks good to me